tstrait/genetic.py

```diff
@@ -0,0 +1,207 @@
+            genotype = np.zeros(self.ts.num_individuals)
+        else:
+            genotype = _traversal_genotype(
+                nodes_individual=self.ts.nodes_individual,
+                left_child_array=tree.left_child_array,
+                right_sib_array=tree.right_sib_array,
+                stack=stack,
+                has_mutation=has_mutation,
+                num_individuals=self.ts.num_individuals,
+                num_nodes=num_nodes,
+            )
+
+        return genotype
+
+    def _run(self):
+        """Computes genetic values of individuals.
+
+        Returns
+        -------
+        pandas.DataFrame
+            Dataframe with simulated genetic value, individual ID, and trait ID.
+        """
+
+        num_ind = self.ts.num_individuals
+        num_trait = np.max(self.trait_df.trait_id) + 1
+        genetic_val_array = np.zeros((num_trait, num_ind))
+        num_nodes = self.ts.num_nodes
+        tree = tskit.Tree(self.ts)
+
+        for data in self.trait_df.itertuples():
+            site = self.ts.site(data.site_id)
+            tree.seek(site.position)
+            individual_genotype = self._individual_genotype(
+                tree=tree,
+                site=site,
+                causal_state=data.causal_state,
+                num_nodes=num_nodes,
+            )
+            genetic_val_array[data.trait_id, :] += (
+                individual_genotype * data.effect_size
+            )
+
+        df = pd.DataFrame(
+            {
+                "trait_id": np.repeat(np.arange(num_trait), num_ind),
+                "individual_id": np.tile(np.arange(num_ind), num_trait),
+                "genetic_value": genetic_val_array.flatten(),
+            }
+        )
+
+        return df
+
+
+def genetic_value(ts, trait_df):
+    """
+    Obtains genetic value from a trait dataframe.
+
+    Parameters
+    ----------
+    ts : tskit.TreeSequence
+        The tree sequence data that will be used in the quantitative trait
+        simulation.
+    trait_df : pandas.DataFrame
+        Trait dataframe.
+
+    Returns
+    -------
+    pandas.DataFrame
+        Pandas dataframe that includes genetic value of individuals in the
+        tree sequence data.
+
+    See Also
+    --------
+    trait_model : Return a trait model, which can be used as `model` input.
+    sim_trait : Return a trait dataframe, whch can be used as a `trait_df` input.
+    GeneticResult : Dataclass object that will be used as an output.
+    sim_env : Genetic value dataframe output can be used as an input to simulate
+        environmental noise.
+
+    Notes
+    -----
+    The `trait_df` input has some requirements that will be noted below.
+
+    1. Columns
+
+    The following columns must be included in `trait_df`:
+
+        * **site_id**: Site IDs that have causal mutation.
+        * **effect_size**: Simulated effect size of causal mutation.
+        * **causal_state**: Derived state of causal mutation.
```

Comment:
This isn't correct - we can have multiple causal mutations. Our model is a causal *allele*, right?

Reply:
Just to be clear, am I correct with saying that we should name the `causal_state` column to be `causal_allele` instead?

Reply:
Yes, that's what we're discussing in #113 

I'm pointing out that the documentation is incorrect here though, because you say causal *mutation*. We can have more than one mutation to the causal allele.